py4cast/datasets/access.py

```diff
@@ -0,0 +1,442 @@
+
+    # (Nx, Ny, 1)
+    geopotential = torch.tensor(grid.geopotential).unsqueeze(2)  # (N_x, N_y, 1)
+    gp_min = torch.min(geopotential)
+    gp_max = torch.max(geopotential)
+    # Rescale geopotential to [0,1]
+    if gp_max != gp_min:
+        geopotential = (geopotential - gp_min) / (gp_max - gp_min)  # (N_x,N_y, 1)
+    else:
+        warnings.warn("Geopotential is constant. Set it to 1")
+        geopotential = geopotential / gp_max
+
+    grid_border_mask = torch.tensor(grid.border_mask).unsqueeze(2)  # (N_x, N_y,1)
+
+    feature_names = []
+    for x in extra_statics:
+        feature_names += x.feature_names
+    state_var = NamedTensor(
+        tensor=torch.cat(
+            [grid_xy, geopotential, grid_border_mask]
+            + [x.tensor for x in extra_statics],
+            dim=-1,
+        ),
+        feature_names=["x", "y", "geopotential", "border_mask"]
+        + feature_names,  # Noms des champs 2D
+        names=["lat", "lon", "features"],
+    )
+    state_var.type_(torch.float32)
+    return state_var
+
+
+ParamConfig = namedtuple(
+    "ParamConfig", "unit level_type long_name grid grib_name grib_param"
+)
+
+
+@dataclass(slots=True)
+class WeatherParam:
+    name: str
+    level: int
+    grid: Grid
+    load_param_info: Callable[[str], ParamConfig]
+    # Parameter status :
+    # input = forcings, output = diagnostic, input_output = classical weather var
+    kind: Literal["input", "output", "input_output"]
+    get_weight_per_level: Callable[[int, str], [float]]
+    level_type: str = field(init=False)
+    long_name: str = field(init=False)
+    unit: str = field(init=False)
+    native_grid: str = field(init=False)
+    grib_name: str = field(init=False)
+    grib_param: str = field(init=False)
+
+    def __post_init__(self):
+        param_info = self.load_param_info(self.name)
+        self.unit = param_info.unit
+        if param_info.level_type in ["heightAboveGround", "meanSea", "surface"]:
+            self.level_type = param_info.level_type
+        else:
+            self.level_type = "isobaricInhPa"
+        self.long_name = param_info.long_name
+        self.native_grid = param_info.grid
+        self.grib_name = param_info.grib_name
+        self.grib_param = param_info.grib_param
+
+    @property
+    def state_weight(self) -> float:
+        """Weight to confer to the param in the loss function"""
+        return self.get_weight_per_level(self.level, self.level_type)
+
+    @property
+    def parameter_name(self) -> str:
+        return f"{self.long_name}_{self.level}_{self.level_type}"
+
+
+@dataclass
+class Stats:
+    fname: Path
+
+    def __post_init__(self):
+        self.stats = torch.load(self.fname, "cpu", weights_only=True)
+
+    def items(self):
+        return self.stats.items()
+
+    def __getitem__(self, shortname: str):
+        return self.stats[shortname]
+
+    def to_list(
+        self,
+        aggregate: Literal["mean", "std", "min", "max"],
```

Comment:
```suggestion
        stat_name: Literal["mean", "std", "min", "max"],
```

aggregate doesn't mean much does it ?

Reply:
it does imho, but clearly less explicit than your proposal